### ISSUE TITLE
PyCon tutorial(s)

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/zero-to-jupyterhub-k8s#271]
Hey all - @yuvipanda brought up the good point that PyCon is coming up, and this might be a great place to highlight all of the awesome work that's been happening in the JupyterHub world. I feel like there'd be value in either giving a talk or a tutorial (or both). Off the top of my head, I feel like we could do any of:

* Zero to JupyterHub tutorial (similar to Yuvi's tutorial at jupytercon)
* non kubernetes jupyterhub tutorial (similar to Min's first tutorial at Jupytercon)
* Zero to BinderHub tutorial
* Overview / vision of the JupyterHub/BinderHub world (maybe similar to the "whirlwind tour" we gave at the MSDSE summit)
* others?

perhaps @willingc has thoughts on what would be most interesting for this audience?

cc @Carreau @aculich @willingc @yuvipanda @minrk

**Comment by Carreau** [comment]
+1 I missed msdse so would be happy to get a second chance to see it !

**Comment by willingc** [comment]
Any of the above topics would be good for talks. As for tutorials, you want to make sure that a significant amount of the time is used as hands on, be something folks would pay for, and reaches a reasonable audience. I think Zero to BinderHub and how to for different repos might be a well scoped tutorial. Ideally, if you can teach the tutorial using minikube without having to have a cloud service that would be best.

Adding @betatim to the discussion too as he attended PyCon last year. Some of his Tour De France analysis would also be appealing demos.

**Comment by Carreau** [comment]
AFAICT Minicube might be a pain.

+1 on a binderhub, by then we may have a federation prototype.

**Comment by betatim** [comment]
> Some of his Tour De France analysis would also be appealing demos.

When you say TdF you mean the plots of some of the stages I made? I'd have to go check I still have the live tracking data ...

Tutorial topics wise: showing people how to use bhub/jhub in a setting at work (close to production env even for interactive work, easy sharing amongst teams, we have spark can we still jhub?, can I execute my binder thing locally?, what about using it for training?, etc) would be something I'd be interested in. This is more how to use/do cool stuff with bhub/jhub than setting it up.

Maybe [this talk proposal](https://public.etherpad-mozilla.org/p/tim-trams-sps17) I wrote for the Swiss Python Summit to talk about somewhat related topics helps. Don't ask me how the talk will flow, I have till February to work that out.

**Comment by willingc** [comment]
> Tutorial topics wise: showing people how to use bhub/jhub in a setting at work (close to production env even for interactive work, easy sharing amongst teams, we have spark can we still jhub?, can I execute my binder thing locally?, what about using it for training?, etc) would be something I'd be interested in. This is more how to use/do cool stuff with bhub/jhub than setting it up.

@betatim nailed what would make a good tutorial ("use/do cool stuff").

**Comment by betatim** [comment]
Random tidbit: A half day tutorial session lasts for 200minutes (in 2017) or 180minutes (2018 CfP), so it isn't actually that long.

**Comment by minrk** [comment]
Zero to BinderHub, plus tips on building binder repos sounds pretty good.

**Comment by choldgraf** [comment]
What if we led with some interesting recipes for Binder repos running off of `mybinder.org`, then we went through Z2BH for the remainder?

**Comment by yuvipanda** [comment]
Looks like we missed the deadline on this one. Next time I guess?

**Comment by choldgraf** [comment]
I thought @Carreau said it was sometime in mid-january, but I think I was mistaken :-(  this one slipped because the deadline ended on thanksgiving weekend and I've been (mostly) checked out spending time w/ family

**Comment by Carreau** [comment]
Talks CFP end in Jan. Tutorial is indeed too late.

**Comment by aculich** [comment]
@Carreau @choldgraf of course it's never too late to submit "late tutorial proposals", right? Worth a shot asking if they'll still accept a submission?

**Comment by willingc** [comment]
For PyCon, late submissions for tutorials are not accepted.

**Comment by choldgraf** [comment]
yeah, we just missed the boat on this one. there will be more pycons! let's focus our efforts on finding other venues to connect w/ people. Also we can still submit a talk to pycon, which at least could help generate interest / excitement.

**Comment by willingc** [comment]
Closing re: tutorial.

**Comment by yuvipanda** [comment]
As a note, I am not submitting any talks to PyCon this year (too many other things going on). Wanted to say that here since I had suggested otherwise earlier!

**Comment by choldgraf** [comment]
Does anybody know about the difference between the conference and the summits? There's a "Python for Education" summit, which seems like a natural place to talk about JupyterHub / Binder

**Comment by willingc** [comment]
CFP closed yesterday for talks, posters, and education summit for PyCon. The Education Summit happens the second day of the tutorials and is a more casual event which focuses on folks working as educators. I submitted a talk to the Ed Summit about using Jupyter Notebooks, JupyterHub, and Binder for STEAM workshops for High School/Middle School outreach. I also submitted a talk proposal and poster for Intermediate Sphinx. Who knows if any will be accepted.
